src/drivers/driver.ts

```diff
@@ -614,10 +615,64 @@ export abstract class CMakeDriver implements vscode.Disposable {
     await this.doPreCleanConfigure();
     this.configRunning = false;
 
-    return this.configure(extra_args, consumer);
+    return this.configure(trigger, extra_args, consumer);
   }
 
-  async configure(extra_args: string[], consumer?: proc.OutputConsumer, withoutCmakeSettings:boolean = false): Promise<number> {
+  async testCompilerVersion(program: string, cwd: string, arg: string | undefined, regexp: RegExp): Promise<string> {
+    const args = [];
+    if (arg) {
+      args.push(arg);
+    }
+    const child = this.executeCommand(program, args, undefined, {silent: true, cwd});
+    try {
+      const result = await child.result;
+      log.debug(localize('command.version.test.return.code', 'Command version test return code {0}', nullableValueToString(result.retc)));
+      // cl.exe version information is printed on stderr when no source files are given
+      const versionLine = (program === "cl") ? result.stderr : result.stdout; //.split("\\n")[0];
+      const match = regexp.exec(versionLine);
+      return match ? match[1] : "error";
+    } catch (e) {
+      const e2: NodeJS.ErrnoException = e;
+      log.debug(localize('compiler.version.return.code', 'Compiler version test return code {0}', nullableValueToString(e2.code)));
+      if (e2.code === 'error') {
+        return e2.code;
+      }
+      throw e;
+    }
+  }
+
+  async getCompilerVersion(compilerPath: string) : Promise<string | undefined> {
+    const compilerName = path.parse(compilerPath).name;
+    const compilerDir = path.parse(compilerPath).dir;
+    let versionSwitch;
+    let versionRegexp;
+    switch (compilerName) {
+      case "cl":
+        // cl does not have a version switch but it outputs the compiler version on stderr
+        // when no source files arguments are given
+        versionRegexp = /.* Compiler Version (.*) for .*/mg;
+        break;
+      case "gcc":
+      case "g++:":
+      case "cpp:":
+      case "c++:":
+        versionSwitch = "--version";
+        versionRegexp = RegExp(`${compilerName} .* (.*)`, "mg");
```

Comment:
The first `.*` should probably be a `[^\s]+` instead.  `.` matches whitespace too and you specifically don't want to match whitespace for this.

Reply:
Not sure. Here is the version output I started from: 
**cpp (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0**
With the expression above, I am looking at match[1] and obtain **9.3.0**
With the suggestion I get this whole thing: **9.3.0-17ubuntu1~20.04) 9.3.0**

Reply:
Inspired by the areas you recommended me to look at, I realized -v is more general version switch and it produces different kind of output than --version, in which case your suggestion is perfect and you'll see in the next update.

Reply:
So if you want to capture the thing after the last space, then you would use `\`${compilerName}.*([^\s]+)$`

---
In reply to: [516683687](https://github.com/Microsoft/vscode-cmake-tools/pull/1503#discussion_r516683687) [](ancestors = 516683687)

Reply:
Oh, but if you're using multiline regex, then I don't know.  I guess you can leave it if it's doing the right thing.